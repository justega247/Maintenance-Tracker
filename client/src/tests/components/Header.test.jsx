import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ConnectedHeader, { Header } from '../../components/Header';

let wrapper;
let props;

const createMockStore = configureMockStore([thunk]);
const store = createMockStore({
  auth: {
    isAuthenticated: true,
    user: {
      id: 2,
    },
  },
});
describe('Header', () => {
  beforeEach(() => {
    props = {
      auth: true,
      userId: 1,
    };
    wrapper = shallow(<Header {...props} />);
  });
  test('should correctly render Header', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render correct header when user is not the admin', () => {
    const props1 = {
      auth: true,
      userId: 2,
    };
    const wrapper1 = shallow(<Header {...props1} />);
    expect(wrapper1).toMatchSnapshot();
  });

  test('should render correct header when user is not logged in', () => {
    const props2 = {
      auth: false,
      userId: 2,
    };
    const wrapper1 = shallow(<Header {...props2} />);
    expect(wrapper1).toMatchSnapshot();
  });

  test('should start the logout action on button click', () => {
    const startLogout = jest.fn();
    const wrapper1 = shallow(<Header auth userId={2} startLogout={startLogout} />);
    wrapper1.find('.logout__link').simulate('click');
    expect(startLogout).toHaveBeenCalled();
  });

  test('should render connected component', () => {
    wrapper = shallow(<ConnectedHeader store={store} {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
