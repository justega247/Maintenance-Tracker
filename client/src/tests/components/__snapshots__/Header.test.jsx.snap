// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should correctly render Header 1`] = `
<div>
  <header
    className="header auth__header"
  >
    <Link
      className="auth__title"
      replace={false}
      to="/admin-dashboard"
    >
      <span
        className="header__name"
      >
        Maintenance Tracker 
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="cogs"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </span>
    </Link>
    <div
      className="nav__div"
    >
      <Link
        className="nav__item"
        replace={false}
        to="/admin-dashboard"
      >
        View Requests
      </Link>
      <Link
        className="nav__item logout__link"
        onClick={[Function]}
        replace={false}
        to="#"
      >
        Logout
      </Link>
    </div>
  </header>
</div>
`;

exports[`Header should render correct header when user is not logged in 1`] = `
<div>
  <header
    className="header"
  >
    <Link
      className="header__title"
      replace={false}
      to="/"
    >
      <span>
        Maintenance Tracker 
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="cogs"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </span>
    </Link>
  </header>
</div>
`;

exports[`Header should render correct header when user is not the admin 1`] = `
<div>
  <header
    className="header auth__header"
  >
    <Link
      className="auth__title"
      replace={false}
      to="/user-dashboard"
    >
      <span
        className="header__name"
      >
        Maintenance Tracker 
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="cogs"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </span>
    </Link>
    <div
      className="nav__div"
    >
      <Link
        className="nav__item"
        replace={false}
        to="/create-request"
      >
        Create Request
      </Link>
      <Link
        className="nav__item"
        replace={false}
        to="/user-dashboard"
      >
        View Requests
      </Link>
      <Link
        className="nav__item logout__link"
        onClick={[Function]}
        replace={false}
        to="#"
      >
        Logout
      </Link>
    </div>
  </header>
</div>
`;
